vm/src/obj/objdict.rs

```diff
@@ -37,92 +38,12 @@ impl PyValue for PyDict {
     }
 }
 
-pub fn get_elements<'a>(obj: &'a PyObjectRef) -> impl Deref<Target = DictContentType> + 'a {
-    obj.payload::<PyDict>().unwrap().entries.borrow()
-}
-
-pub fn get_mut_elements<'a>(obj: &'a PyObjectRef) -> impl DerefMut<Target = DictContentType> + 'a {
-    obj.payload::<PyDict>().unwrap().entries.borrow_mut()
-}
-
-pub fn set_item(
-    dict: &PyObjectRef,
-    _vm: &VirtualMachine,
-    needle: &PyObjectRef,
-    value: &PyObjectRef,
-) {
-    // TODO: use vm to call eventual __hash__ and __eq__methods.
-    let mut elements = get_mut_elements(dict);
-    set_item_in_content(&mut elements, needle, value);
-}
-
-pub fn set_item_in_content(
-    elements: &mut DictContentType,
-    needle: &PyObjectRef,
-    value: &PyObjectRef,
-) {
-    // XXX: Currently, we only support String keys, so we have to unwrap the
-    // PyObject (and ensure it is a String).
-
-    // TODO: invoke __hash__ function here!
-    let needle_str = objstr::get_value(needle);
-    elements.insert(needle_str, (needle.clone(), value.clone()));
-}
-
 pub fn get_key_value_pairs(dict: &PyObjectRef) -> Vec<(PyObjectRef, PyObjectRef)> {
-    let dict_elements = get_elements(dict);
-    get_key_value_pairs_from_content(&dict_elements)
-}
-
-pub fn get_key_value_pairs_from_content(
-    dict_content: &DictContentType,
-) -> Vec<(PyObjectRef, PyObjectRef)> {
-    let mut pairs: Vec<(PyObjectRef, PyObjectRef)> = Vec::new();
-    for (_str_key, pair) in dict_content.iter() {
-        let (key, obj) = pair;
-        pairs.push((key.clone(), obj.clone()));
-    }
-    pairs
-}
-
-pub fn get_item(dict: &PyObjectRef, key: &PyObjectRef) -> Option<PyObjectRef> {
-    let needle_str = objstr::get_value(key);
-    get_key_str(dict, &needle_str)
-}
-
-// Special case for the case when requesting a str key from a dict:
-pub fn get_key_str(dict: &PyObjectRef, key: &str) -> Option<PyObjectRef> {
-    let elements = get_elements(dict);
-    content_get_key_str(&elements, key)
-}
-
-/// Retrieve a key from dict contents:
-pub fn content_get_key_str(elements: &DictContentType, key: &str) -> Option<PyObjectRef> {
-    // TODO: let hash: usize = key;
-    match elements.get(key) {
-        Some(v) => Some(v.1.clone()),
-        None => None,
-    }
-}
-
-pub fn contains_key_str(dict: &PyObjectRef, key: &str) -> bool {
-    let elements = get_elements(dict);
-    content_contains_key_str(&elements, key)
-}
-
-pub fn content_contains_key_str(elements: &DictContentType, key: &str) -> bool {
-    // TODO: let hash: usize = key;
-    elements.get(key).is_some()
-}
-
-/// Take a python dictionary and convert it to attributes.
-pub fn py_dict_to_attributes(dict: &PyObjectRef) -> PyAttributes {
-    let mut attrs = PyAttributes::new();
-    for (key, value) in get_key_value_pairs(dict) {
-        let key = objstr::get_value(&key);
-        attrs.insert(key, value);
-    }
-    attrs
+    dict.payload::<PyDict>()
```

Comment:
can't this take a `PyDictRef` too?
(may be a followup patch)

Reply:
The are code paths that support both. A lot of callers have ```PyObjectRef```. I should change the callers to cast to a PyDictRef, and then this is just a simple method on ```PyDictRef```.

I've added to my clean-up list.